### PR TITLE
Choreographer upgrade rbocsi

### PR DESCRIPTION
**Pull request by rbocsi**

### CODE REVIEW

File: choreographer/src/main/java/eu/arrowhead/core/choreographer/ChoreographerNotifyController.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ 
+    //-------------------------------------------------------------------------------------------------
+    @ApiOperation(value = "Notify the Choreographer that a step is done in a session.", tags = { CoreCommonConstants.SWAGGER_TAG_CLIENT })
+    @ApiResponses(value = {
+            @ApiResponse(code = HttpStatus.SC_CREATED, message = STEP_DONE_HTTP_200_MESSAGE),
+            @ApiResponse(code = HttpStatus.SC_BAD_REQUEST, message = STEP_DONE_HTTP_400_MESSAGE),
+            @ApiResponse(code = HttpStatus.SC_UNAUTHORIZED, message = CoreCommonConstants.SWAGGER_HTTP_401_MESSAGE),
+            @ApiResponse(code = HttpStatus.SC_INTERNAL_SERVER_ERROR, message = CoreCommonConstants.SWAGGER_HTTP_500_MESSAGE)
+    })
+    @PostMapping(path = CommonConstants.OP_CHOREOGRAPHER_NOTIFY_STEP_DONE, consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (borditamas) [inline choreographer/src/main/java/eu/arrowhead/core/choreographer/ChoreographerNotifyController.java:71]: no need for `produces`

File: choreographer/src/main/java/eu/arrowhead/core/choreographer/ChoreographerNotifyController.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    //-------------------------------------------------------------------------------------------------
+    @ApiOperation(value = "Notify the Choreographer that an error happened during the execution of a step in a session.", tags = { CoreCommonConstants.SWAGGER_TAG_CLIENT })
+    @ApiResponses(value = {
+            @ApiResponse(code = HttpStatus.SC_CREATED, message = STEP_DONE_HTTP_200_MESSAGE),
+            @ApiResponse(code = HttpStatus.SC_BAD_REQUEST, message = STEP_DONE_HTTP_400_MESSAGE),
+            @ApiResponse(code = HttpStatus.SC_UNAUTHORIZED, message = CoreCommonConstants.SWAGGER_HTTP_401_MESSAGE),
+            @ApiResponse(code = HttpStatus.SC_INTERNAL_SERVER_ERROR, message = CoreCommonConstants.SWAGGER_HTTP_500_MESSAGE)
+    })
+    @PostMapping(path = CommonConstants.OP_CHOREOGRAPHER_EXECUTOR_NOTIFY_STEP_ERROR, consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (borditamas) [inline choreographer/src/main/java/eu/arrowhead/core/choreographer/ChoreographerNotifyController.java:87]: no need for `produces`

File: choreographer/src/main/java/eu/arrowhead/core/choreographer/ChoreographerNotifyController.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    @ResponseBody public void notifyStepDone(@RequestBody final ChoreographerSessionRunningStepDataDTO request) {
+        logger.debug("notifyStepDone started...");
+        logger.debug("Sending message to session-step-done.");
+        jmsTemplate.convertAndSend("session-step-done", request);
+    }
+
+    //-------------------------------------------------------------------------------------------------
+    @ApiOperation(value = "Notify the Choreographer that an error happened during the execution of a step in a session.", tags = { CoreCommonConstants.SWAGGER_TAG_CLIENT })
+    @ApiResponses(value = {
+            @ApiResponse(code = HttpStatus.SC_CREATED, message = STEP_DONE_HTTP_200_MESSAGE),
```
Comment (borditamas) [inline choreographer/src/main/java/eu/arrowhead/core/choreographer/ChoreographerNotifyController.java:82]: should be STEP_ERROR... messages

File: choreographer/src/main/java/eu/arrowhead/core/choreographer/validation/ActionCircleDetector.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+	//-------------------------------------------------------------------------------------------------
+	public boolean hasCircle(final ChoreographerPlanRequestDTO plan) {
+		logger.debug("hasCircle started...");
+		
+		final Set<String> checkedActions = new HashSet<>();
+		ChoreographerActionRequestDTO action = findAction(plan.getActions(), plan.getFirstActionName());
+		checkedActions.add(action.getName());
+		
+		while (!Utilities.isEmpty(action.getNextActionName())) {
```
Comment (borditamas) [inline choreographer/src/main/java/eu/arrowhead/core/choreographer/validation/ActionCircleDetector.java:34]: action can be null due to `findAction`

File: choreographer/src/main/java/eu/arrowhead/core/choreographer/database/service/ChoreographerPlanDBService.java [hunk 5]
```diff
@@ -0,0 +1,616 @@
...
+	
+	//-------------------------------------------------------------------------------------------------
+	@Transactional(rollbackFor = ArrowheadException.class)
+    public void removePlanEntryById(final long id) {
+        logger.debug("removePlanEntryById started...");
+
+        try {
+        	final Optional<ChoreographerPlan> planOpt = choreographerPlanRepository.findById(id);
+            if (planOpt.isEmpty()) {
+                throw new InvalidParameterException("Plan with id of '" + id + "' doesn't exist!");
```
Comment (borditamas) [inline choreographer/src/main/java/eu/arrowhead/core/choreographer/database/service/ChoreographerPlanDBService.java:177]: why not just return?

Comment (rbocsi) [reply choreographer/src/main/java/eu/arrowhead/core/choreographer/database/service/ChoreographerPlanDBService.java:177]: That's how remove works for example in ServiceRegistryDBService.

**Review by borditamas — APPROVED** [review]
but made some comments

File: choreographer/src/main/java/eu/arrowhead/core/choreographer/database/service/ChoreographerPlanDBService.java (hunk 5 above)
Comment (borditamas) [reply choreographer/src/main/java/eu/arrowhead/core/choreographer/database/service/ChoreographerPlanDBService.java:177]: than maybe I should follow this regarding to the executros too

**Pull request merged**